src/d3a/models/market/blockchain_interface.py

```diff
@@ -95,14 +99,17 @@ def handle_blockchain_trade_event(self, offer, buyer, original_offer, residual_o
         return str(uuid.uuid4()), residual_offer
 
     @retry_function(max_retries=3)
-    def track_trade_event(self, trade):
+    def track_trade_event(self, time_slot, trade):
 
         call_params = {
+            'simulation_id': str(self.simulation_id),
+            'market_id': str(self.market_id),
+            'market_slot': datetime.timestamp(time_slot),
             'trade_id': trade.id,
             'buyer': ALICE_STASH_ADDRESS,
             'seller': BOB_STASH_ADDRESS,
-            'energy': trade.offer.energy,
-            'rate': trade.offer.energy_rate
+            'energy': int(trade.offer.energy),
+            'rate': int(trade.offer.energy_rate)
```

Comment:
Why do you need to cast these as int ? Especially for the energy value, this usually is between 0-1, which means that this casting would immediately convert it to zero. I would omit it actually.

Reply:
The reason I need to cast them to `int`is due to pallet don't accept floating point data type.
If you agree I will do the conversion of energy from kWh to Wh so integer casting doesn't round it to zero.